test/integration/testpaths/testpath_attach_disk_zwps.py

```diff
@@ -0,0 +1,209 @@
+                cls.apiclient,
+                cls.testdata["service_offering"],
+                tags=ZONETAG1
+            )
+            cls._cleanup.append(cls.service_offering_zone1)
+
+            # Create Disk offering
+            cls.disk_offering = DiskOffering.create(
+                cls.apiclient,
+                cls.testdata["disk_offering"]
+            )
+
+            cls._cleanup.append(cls.disk_offering)
+
+        except Exception as e:
+            cls.tearDownClass()
+            raise e
+        return
+
+    @classmethod
+    def tearDownClass(cls):
+        try:
+            cleanup_resources(cls.apiclient, cls._cleanup)
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+
+    def setUp(self):
+        self.apiclient = self.testClient.getApiClient()
+        self.dbclient = self.testClient.getDbConnection()
+        self.cleanup = []
+
+    def tearDown(self):
+        try:
+            for storagePool in self.pools:
+                StoragePool.update(self.apiclient, id=storagePool.id, tags="")
+
+            if hasattr(self, "data_volume_created"):
+                data_volumes_list = Volume.list(
+                    self.userapiclient,
+                    id=self.data_volume_created.id,
+                    virtualmachineid=self.vm.id
+                )
+                if data_volumes_list:
+                    self.vm.detach_volume(
+                        self.userapiclient,
+                        data_volumes_list[0]
+                    )
+
+                status = validateList(data_volumes_list)
+                self.assertEqual(
+                    status[0],
+                    PASS,
+                    "DATA Volume List Validation Failed")
+
+            cleanup_resources(self.apiclient, self.cleanup)
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+        return
+
+    @attr(tags=["basic", "advanced"], required_hardware="true")
+    def test_01_attach_datadisk_to_vm_on_zwps(self):
+        """ Attach Data Disk To VM on ZWPS
+            1.  Check if zwps storage pool exists.
+            2.  Adding tag to zone wide primary storage
+            3.  Launch a VM on ZWPS
+            4.  Attach data disk to vm which is on zwps.
+            5.  Verify disk is attached.
+        """
+
+        # Step 1
+        assert len(list(storagePool for storagePool in self.pools
+                        if storagePool.scope == "ZONE")) >= 1,\
+            "There must be at least one zone wide\
+                storage pools available in the setup"
+
+        # Adding tags to Storage Pools
+        zone_no = 1
+        for storagePool in self.pools:
+            if storagePool.scope == "ZONE":
+                StoragePool.update(
+                    self.apiclient,
+                    id=storagePool.id,
+                    tags=['zwps' + repr(zone_no)])

```

Comment:
Updated the testcase:
     -Skip testcase instead for failing it for no Zone wide primary storage found
     -Replace hard-coded values by code for tag.